File: plugins/dotnet/robocode.dotnet.napi/src/net/sf/robocode/security/SafeComponent.cs
using System.Runtime.CompilerServices;
using IKVM.Attributes;
using java.awt;

namespace net.sf.robocode.security
{
    public class SafeComponent : Component
    {
        private static Component safeEventComponent;

        [MethodImpl(MethodImplOptions.NoInlining)]
        public static void __<clinit>()
        {
        }

        [MethodImpl(MethodImplOptions.NoInlining), LineNumberTable(new byte[] {0x9f, 0xa9, 0x6a, 0x8a})]
        public static Component getSafeEventComponent()
        {
            if (safeEventComponent == null)
            {
                safeEventComponent = new SafeComponent();
            }
            return safeEventComponent;
        }
    }
}

File: robocode.api/src/main/java/net/sf/robocode/security/SafeComponent.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package net.sf.robocode.security;


import java.awt.*;


/**
 * @author Pavel Savara (original)
 */
@SuppressWarnings("serial")
public class SafeComponent extends Component {
	// Dummy component used to preventing robots in accessing the real source component
	private static Component safeEventComponent;

	public static Component getSafeEventComponent() {
		if (safeEventComponent == null) {
			safeEventComponent = new SafeComponent();
		}
		return safeEventComponent;
	}
}
